tests/DCPS/Compiler/optional/CMakeLists.txt

```
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.8...3.27)
+project(opendds_compiler_optional CXX)
+enable_testing()
+
+find_package(OpenDDS REQUIRED)
+include(opendds_testing)
+
+set(target_prefix "${PROJECT_NAME}_")
+set(dst ${CMAKE_CURRENT_BINARY_DIR})
+set(opendds_libs
+  OpenDDS::Dcps
+)
+
+# IDL Library
+set(idl "${target_prefix}idl")
+add_library(${idl})
+opendds_target_sources(${idl} PUBLIC "optional.idl" OPENDDS_IDL_OPTIONS "-Lc++11")
+target_link_libraries(${idl} PUBLIC OpenDDS::Dcps)
+set(idl_dir "${CMAKE_CURRENT_BINARY_DIR}")
+
+set(optional "${target_prefix}exe")
+add_executable(${optional}
+  "main.cpp"
+)
+set_target_properties(${optional} PROPERTIES
+  OUTPUT_NAME optional_annotation
+  RUNTIME_OUTPUT_DIRECTORY "${dst}"
+)
+target_link_libraries(${optional} ${opendds_libs} ${idl})
+
+configure_file(run_test.pl . COPYONLY)
+opendds_add_test(COMMAND perl run_test.pl cpp11 EXTRA_LIB_DIRS "${idl_dir}")
```

```suggestion
add_executable(${PROJECT_NAME}  "main.cpp")
set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME optional_annotation)
opendds_target_sources(${PROJECT_NAME} PUBLIC "optional.idl" OPENDDS_IDL_OPTIONS "-Lc++11")
target_link_libraries(${PROJECT_NAME} OpenDDS::Dcps)

configure_file(run_test.pl . COPYONLY)
opendds_add_test(COMMAND perl run_test.pl)
```

No need for a separate library and so all of this can be simplified.

---

The idea would be to add some proper tests, which would required the idl library, so I left it here for now

---

The MPC test doesn't have a separate IDL library and even if it did having a separate library when it's not needed doesn't make it more proper.